implementations/rust/ockam/ockam_command/src/upgrade.rs

```diff
@@ -5,42 +5,58 @@ use ockam_core::env::get_env_with_default;
 use serde::Deserialize;
 use std::env;
 
-#[derive(Deserialize)]
-pub struct UpgradeFile {
-    #[serde(default = "default_upgrade_message")]
-    pub upgrade_message: String,
-    #[serde(default = "default_upgrade_message_macos")]
-    pub upgrade_message_macos: String,
-}
+const RELEASE_TAG_NAME_PREFIX: &str = "ockam_v";
 
-fn default_upgrade_message() -> String {
-    "Check out the latest release at https://github.com/build-trust/ockam/releases".to_string()
+#[derive(Deserialize, Debug)]
+struct Release {
+    tag_name: String,
+    html_url: String,
 }
 
-fn default_upgrade_message_macos() -> String {
-    "Run the following command to upgrade the Ockam Command: 'brew install build-trust/ockam/ockam'"
-        .to_string()
+impl Release {
+    fn version(&self) -> Option<&str> {
+        self.tag_name
+            .split_once(RELEASE_TAG_NAME_PREFIX)
+            .map(|(_, version)| version)
+    }
 }
 
 pub fn check_if_an_upgrade_is_available(global_args: &GlobalArgs) {
     if upgrade_check_is_disabled() || global_args.test_argument_parser {
         return;
     }
-    let url = format!(
-        "https://github.com/build-trust/ockam/releases/download/ockam_v{}/upgrade.json",
-        crate_version!()
-    );
-    if let Ok(r) = reqwest::blocking::get(url) {
-        if let Ok(f) = r.json::<UpgradeFile>() {
-            let terminal = Terminal::from(global_args);
+
+    // All GitHub API requests must include a valid `User-Agent` header.
+    // See https://docs.github.com/en/rest/using-the-rest-api/getting-started-with-the-rest-api?apiVersion=2022-11-28#user-agent
+    let client = reqwest::blocking::Client::builder()
+        .user_agent("ockam")
+        .build()
+        .unwrap();
+
+    let Ok(latest_release) = client
+        .get("https://api.github.com/repos/build-trust/ockam/releases/latest")
+        .send()
+        .and_then(|resp| resp.json::<Release>())
+    else {
+        return;
+    };
+
+    let latest_version = latest_release.version().unwrap();
+    if crate_version!() != latest_version {
+        let terminal = Terminal::from(global_args);
+
+        terminal
+            .write_line(fmt_info!(
+                "A newer version (v{}) is now available at {}",
+                latest_version,
+                latest_release.html_url
+            ))
+            .unwrap();
```

Comment:
I would split this message in two lines:

```
A new version is now available: v{}
You can download it at: {}
```

You can also colorize the version and HTML texts as we do here to highlight them:

https://github.com/build-trust/ockam/blob/ab931e57657aeb5d812cb381272992d60377c4ab/implementations/rust/ockam/ockam_command/src/node/stop.rs#L104-L114